Feature is not parsed with \|

Для версии 11.2.0
Пример. в фиче есть такой кусок (должен быть экранирующий слеш  \\|)

* отправить через SOAP "перевод частному лицу"
  | данные1   | Autotest 22245 Test |
  | данные2   | \\|pdf\\|pdf                  |
  | данные3   | -172820                    |
 
Получаем ошибку при сборке
cucumber.runtime.CucumberException: gherkin.ParserException$CompositeParserException: Parser errors:
(16:7): inconsistent cell count within the table
(20:7): inconsistent cell count within the table
(24:7): inconsistent cell count within the table
(28:7): inconsistent cell count within the table
                at cucumber.runtime.FeatureBuilder.parse(FeatureBuilder.java:60)
                at ru.sbtqa.tag.pagefactory.utils.GherkinSerializer.lambda$0(GherkinSerializer.java:62)
                at java.util.ArrayList.forEach(ArrayList.java:1249)
                at ru.sbtqa.tag.pagefactory.utils.GherkinSerializer.reserializeFeatures(GherkinSerializer.java:43)
                at ru.sbtqa.tag.pagefactory.aspects.FragmentsAspect.replaceSteps(FragmentsAspect.java:45)
                at cucumber.runtime.RuntimeOptions.cucumberFeatures(RuntimeOptions.java:311)
                at cucumber.api.junit.Cucumber.<init>(Cucumber.java:81)
                at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
                at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
                at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
                at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
                at org.junit.internal.builders.AnnotatedBuilder.buildRunner(AnnotatedBuilder.java:104)
                at org.junit.internal.builders.AnnotatedBuilder.runnerForClass(AnnotatedBuilder.java:86)
                at org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:59)
                at org.junit.internal.builders.AllDefaultPossibilitiesBuilder.runnerForClass(AllDefaultPossibilitiesBuilder.java:26)
                at org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:59)
                at org.junit.internal.requests.ClassRequest.getRunner(ClassRequest.java:33)
                at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:49)
                at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
                at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
                at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
Caused by: gherkin.ParserException$CompositeParserException: Parser errors:
(16:7): inconsistent cell count within the table
(20:7): inconsistent cell count within the table
(24:7): inconsistent cell count within the table
(28:7): inconsistent cell count within the table
                at gherkin.Parser.parse(Parser.java:143)
                at gherkin.Parser.parse(Parser.java:118)
                at gherkin.Parser.parse(Parser.java:114)
                at cucumber.runtime.FeatureBuilder.parse(FeatureBuilder.java:56)
                ... 20 more

Так это невалидная фича. 

Отправлено с iPhone

> 24 апр. 2019 г., в 22:09, andriuha1987 <notifications@github.com> написал(а):
> 
> Для версии 11.2.0
> Пример. в фиче есть такой кусок (должен быть экранирующий слеш |)
> 
> отправить через SOAP "перевод частному лицу"
> | данные1 | Autotest 22245 Test |
> | данные2 | |pdf|pdf |
> | данные3 | -172820 |
> Получаем ошибку при сборке
> cucumber.runtime.CucumberException: gherkin.ParserException$CompositeParserException: Parser errors:
> (16:7): inconsistent cell count within the table
> (20:7): inconsistent cell count within the table
> (24:7): inconsistent cell count within the table
> (28:7): inconsistent cell count within the table
> 
>             at cucumber.runtime.FeatureBuilder.parse(FeatureBuilder.java:60)
>             at ru.sbtqa.tag.pagefactory.utils.GherkinSerializer.lambda$0(GherkinSerializer.java:62)
>             at java.util.ArrayList.forEach(ArrayList.java:1249)
>             at ru.sbtqa.tag.pagefactory.utils.GherkinSerializer.reserializeFeatures(GherkinSerializer.java:43)
>             at ru.sbtqa.tag.pagefactory.aspects.FragmentsAspect.replaceSteps(FragmentsAspect.java:45)
>             at cucumber.runtime.RuntimeOptions.cucumberFeatures(RuntimeOptions.java:311)
>             at cucumber.api.junit.Cucumber.<init>(Cucumber.java:81)
>             at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
>             at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
>             at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
>             at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
>             at org.junit.internal.builders.AnnotatedBuilder.buildRunner(AnnotatedBuilder.java:104)
>             at org.junit.internal.builders.AnnotatedBuilder.runnerForClass(AnnotatedBuilder.java:86)
>             at org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:59)
>             at org.junit.internal.builders.AllDefaultPossibilitiesBuilder.runnerForClass(AllDefaultPossibilitiesBuilder.java:26)
>             at org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:59)
>             at org.junit.internal.requests.ClassRequest.getRunner(ClassRequest.java:33)
>             at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:49)
>             at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
>             at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
>             at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
> Caused by: gherkin.ParserException$CompositeParserException: Parser errors:
> (16:7): inconsistent cell count within the table
> (20:7): inconsistent cell count within the table
> (24:7): inconsistent cell count within the table
> (28:7): inconsistent cell count within the table
> at gherkin.Parser.parse(Parser.java:143)
> at gherkin.Parser.parse(Parser.java:118)
> at gherkin.Parser.parse(Parser.java:114)
> at cucumber.runtime.FeatureBuilder.parse(FeatureBuilder.java:56)
> ... 20 more
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.

--- a/page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/utils/GherkinSerializer.java
+++ b/page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/utils/GherkinSerializer.java
@@ -143,7 +143,9 @@ public class GherkinSerializer {
 
     private void buildTableRow(TableRow tableRow) {
         List<String> collect = tableRow.getCells().stream()
-                .map(TableCell::getValue).collect(Collectors.toList());
+                .map(tableCell -> tableCell.getValue()
+                        .replaceAll("\\|", "\\\\|"))
+                .collect(Collectors.toList());
         tab(2);
         space(2);
         builder.append("|").append(String.join("|", collect)).append("|");

